### PR TITLE
[Network] Make DNS Zone record imports relative

### LINKED ISSUE #2752: DNS zone import creates absolute names instead of relative
### Environment summary

**Install Method:** `curl -L https://aka.ms/InstallAzureCli | bash`

**CLI Version:** 
```
azure-cli (2.0.2)

acr (2.0.0)
acs (2.0.2)
appservice (0.1.2)
batch (2.0.0)
cloud (2.0.0)
component (2.0.0)
configure (2.0.2)
container (0.1.2)
core (2.0.2)
documentdb (0.1.2)
feedback (2.0.0)
find (0.0.1b1)
iot (0.1.2)
keyvault (2.0.0)
lab (0.0.1)
monitor (0.0.1)
network (2.0.2)
nspkg (2.0.0)
profile (2.0.2)
redis (0.1.1b3)
resource (2.0.2)
role (2.0.1)
sql (2.0.0)
storage (2.0.2)
vm (2.0.2)

Python (Linux) 2.7.13 (default, Jan 13 2017, 10:15:16) 
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

**OS Version:** Linux Fedora 24 (4.9.13-101.fc24.x86_64)

**Shell Type:** bash

### Description

When running a zone import e.g.:

```
az network dns zone import -g my-rg -n mydomain.com -f mydomain-zonefile.txt
```

where `mydomain-zonefile.txt` has the following content:

```
@     600     IN      SOA     ignored.com. hostmaster.mydomain.com. (
                                2017040400
                                28800
                                7200
                                604800
                                600
                                )

test    3600    IN      A       2.2.2.2
```

The relative domain record "test" is imported into Azure DNS as absolute value `test.mydomain.com` instead of relative value `test`, meaning that lookup `test.mydomain.com.mydomain.com` works and `test.mydomain.com` fails.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #2752.  Note that issue #2824 remains and can only be addressed by a service side fix.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] <del>Each new command has a test.</del>

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-network/tests/test_network_commands.py [hunk 1]
```diff
@@ -1407,7 +1407,7 @@ def body(self):
 class NetworkZoneImportExportTest(ResourceGroupVCRTestBase):
 
     def __init__(self, test_method):
-        super(NetworkZoneImportExportTest, self).__init__(__file__, test_method, resource_group='cli_dns_zone_import_export')
+        super(NetworkZoneImportExportTest, self).__init__(__file__, test_method, resource_group='cli_dns_zone_import_export', debug=True)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/tests/test_network_commands.py]: nit: debug true here.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -1884,8 +1884,20 @@ def import_zone(resource_group_name, zone_name, file_name):
...
+                    # Workaround for issue #2824
+                    relative_record_set_name = record_set_name.rstrip('.')
+                    if not relative_record_set_name.endswith(origin):
+                        logger.warning(
+                            'Cannot import %s. Only records relative to origin may be '
+                            'imported at this time. Skipping...', relative_record_set_name)
+                        continue
+
+                    if record_set_type != 'soa' and relative_record_set_name != origin:
+                        relative_record_set_name = record_set_name[:-(len(origin) + 2)]
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: what's `-(len(origin) + 2)` for?

Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: Part of the workaround to strip the origin back off of the name. The old implementation was to use absolute names. If/when the service fixes itself to allow this, I will just remove this workaround and import all records as absolute references (which the DNS import spec says it should support).

File: src/command_modules/azure-cli-network/tests/test_network_commands.py (hunk 1 above)
Comment (tjprescott) [inline src/command_modules/azure-cli-network/tests/test_network_commands.py]: Noooooooooooooooooo! ☹️

Comment (tjprescott) [inline src/command_modules/azure-cli-network/tests/test_network_commands.py]: Removed. 🕺

**Pull request merged**